DNN Platform/Modules/DnnExportImport/Components/Services/PackagesExportService.cs

```diff
@@ -173,42 +211,13 @@ private void ProcessImportModulePackages(ImportDto importDto)
                     {
                         foreach (var exportPackage in exportPackages)
                         {
-                            try
-                            {
-                                var filePath = Path.Combine(tempFolder, exportPackage.PackageFileName);
-                                if (!File.Exists(filePath))
-                                {
-                                    continue;
-                                }
-
-                                var packageType = exportPackage.PackageType;
-                                var packageName = exportPackage.PackageName;
-                                var version = exportPackage.Version;
-
-                                var existPackage = PackageController.Instance.GetExtensionPackage(Null.NullInteger,
-                                    p => p.PackageType == packageType && p.Name == packageName);
-                                if (existPackage != null &&
-                                    (existPackage.Version > version ||
-                                     (existPackage.Version == version &&
-                                      importDto.CollisionResolution == CollisionResolution.Ignore)))
-                                {
-                                    Result.AddLogEntry("Import Package ignores",
-                                        $"{packageName} has higher version {existPackage.Version} installed, ignore import it");
-                                    continue;
-                                }
-
-                                InstallPackage(filePath);
-                                Result.AddLogEntry("Import Package completed", $"{packageName} version: {version}");
-                                CheckPoint.ProcessedItems++;
-                                CheckPoint.Progress = CheckPoint.ProcessedItems * 100.0 / exportPackages.Count;
-                                CheckPointStageCallback(this); // just to update the counts without exit logic
-                            }
-                            catch (Exception ex)
-                            {
-                                Result.AddLogEntry("Import Package error",
-                                    $"{exportPackage.PackageName} : {exportPackage.Version} - {ex.Message}");
-                                Logger.Error(ex);
-                            }
+
+                            ProcessImportModulePackage(exportPackage, tempFolder, importDto.CollisionResolution);
```

Comment:
Remove tempFolder parameter from the method you are adding pass combined path instead. Also instead of exportPackage pass packageType, packageName, packageVersion; Lastly convert collisionResolution to a boolean called ignoreCollisionResolution.

var filePath = Path.Combine(tempFolder, exportPackage.PackageFileName);
if (File.Exists(filePath))
{
    var packageType = exportPackage.PackageType;
    var packageName = exportPackage.PackageName;
    var packageVersion = exportPackage.Version;
    var ignoreCollisionResolution = collisionResolution == CollisionResolution.Ignore;
    ProcessImportModulePackage(filePath, packageType, packageName, packageVersion, ignoreCollisionResolution);
}
else
{
    LogDebug or Info or Warning for package being skipped as the file does not exist.
}

Reply:
Hi @Yabgu ,

I am trying to avoid the code refactor here. If you observed the code change, I am trying to move the code to a new method because I want to change continue to return without modifying if else statement logic. This is the best to ensure the existing code logic is exactly the same and to add the ProcessedItems count only. Therefore, I am inclined to refactor here.